flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -88,13 +347,280 @@ def add_wind_forcing(
         const_dir : float, optional
             direction of time-invariant wind forcing [deg], by default None
         """
-        self.sf_model.setup_wind_forcing(
-            timeseries=timeseries,
-            magnitude=const_mag,
-            direction=const_dir,
+        if isinstance(forcing, WindConstant):
+            self._model.setup_wind_forcing(
+                timeseries=None,
+                const_mag=forcing.speed,
+                const_dir=forcing.direction,
+            )
+        elif isinstance(forcing, WindTimeSeries):
+            self._model.setup_wind_forcing(
+                timeseries=forcing.path, const_mag=None, const_dir=None
+            )
+        elif isinstance(forcing, WindFromModel):
+            self._model.setup_wind_forcing_from_grid(wind=forcing.path)
+        else:
+            self._logger.warning(
+                f"Unsupported wind forcing type: {forcing.__class__.__name__}"
+            )
+            return
+
+    def _add_forcing_rain(self, forcing: IRainfall):
+        """Add spatially constant rain forcing to sfincs model. Use timeseries or a constant magnitude.
+
+        Parameters
+        ----------
+        timeseries : Union[str, os.PathLike], optional
+            path to file of timeseries file (.csv) which has two columns: time and precipitation, by default None
+        const_intensity : float, optional
+            time-invariant precipitation intensity [mm/hr], by default None
+        """
+        if isinstance(forcing, RainfallConstant):
+            self._model.setup_precip_forcing(
+                timeseries=None,
+                magnitude=forcing.intensity,
+            )
+        elif isinstance(forcing, RainfallSynthetic):
+            self._model.setup_precip_forcing(
+                timeseries=forcing.path,
```

Comment:
Good point, I was thinking that we write all forcings to csv files in the function `event.process()` since we need to combine tide and surge for synthetic, run offshore/download wl for
 historical/hurricanes, which then will affect wind, rainfall etc as you mentioned.
 
So Im thinking we either write everything (every timeseries) to csv files and update these with the information needed. This could be nice since the data is then stored on disk, and we can easliy read it, change it and save it again.

Alternatively, we create functions to compute the data on the fly based on the event model, then we can keep the data in memory and write it directly to the sfincsmodel without storing it. The latter will be probably be faster (although im not sure how much), but a bit harder to implement and debug.